packages/core/lib/debug/printer.js

```diff
@@ -209,10 +210,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+    // printout 3 previous instructions
+    if (currentIndex - previousInstructions + 1 > 0) {
+      this.config.logger.log("...");
+    }
+    for (let i = currentIndex - previousInstructions; i < currentIndex; i++) {
+      if (i >= 0) {
+        this.config.logger.log(DebugUtils.formatInstruction(instructions[i]));
+      }
+    }
+
+    // printout current instruction
+    this.config.logger.log(DebugUtils.formatCurrentInstruction(instruction));
+
+    // printout 3 upcoming instructions
+    for (let i = currentIndex + 1; i <= currentIndex + upcomingInstructions; i++) {
+      if (i >= instructions.length) break;
```

Comment:
Really minor thing, but instead of using `break`, you can use `Math.min`. :)  Feel free to ignore this, the current way is also good; just pointing out the opportunity if you want to make this cleaner. :)

Reply:
changes made